scipy/optimize/_dual_annealing.py

```diff
@@ -625,6 +626,14 @@ def dual_annealing(func, bounds, args=(), maxiter=1000,
     if x0 is not None and not len(x0) == len(bounds):
         raise ValueError('Bounds size does not match x0')
 
+    minimizer_kwargs = {} if minimizer_kwargs is None else minimizer_kwargs
+    if 'jac' in minimizer_kwargs:
+        message = ("Use of `minimize` keyword argument `jac` is deprecated "
+                   "and replaced by `grad`. Support for `jac` will be "
+                   "removed in SciPy 1.13.0.")
+        warnings.warn(message, DeprecationWarning)
+        minimizer_kwargs['grad'] = minimizer_kwargs.pop('jac')
```

Comment:
I might have been able to avoid this and some other `jac` -> `grad` changes by changing 
```python3
self.jac= self.kwargs.get('jac', None)
```
to 
```python3
self.jac= self.kwargs.get('jac', None) or self.kwargs.get('grad', None)
```
I suppose I thought it would be good to go ahead and purge use of `jac` from `dual_annealing`.